src/algebra/quaternion.lean

```diff
@@ -262,6 +262,12 @@ end
 @[norm_cast, simp] lemma coe_mul : ((x * y : R) : ℍ[R, c₁, c₂]) = x * y :=
 (algebra_map R ℍ[R, c₁, c₂]).map_mul x y
 
+@[norm_cast, simp] lemma coe_pow (n : ℕ) : (↑(x ^ n) : ℍ[R, c₁, c₂]) = ↑x ^ n :=
+(algebra_map R ℍ[R, c₁, c₂]).map_pow x n
+
+@[norm_cast] lemma coe_smul (r x : R) : (↑(r • x) : ℍ[R, c₁, c₂]) = r • ↑x :=
+(algebra.linear_map R ℍ[R, c₁, c₂]).map_smul r x
```

Comment:
Repeating [this comment](https://github.com/leanprover-community/mathlib/pull/18349#discussion_r1102066213) here:
Any reason not to have this simp?

Reply:
Because this clashes with the `smul_coe` lemma below. This lemma makes more sense after #18382, at which point it can probably become `simp`.

Reply:
I will remove this from this PR entirely, and leave it till that one.